app/services/applicable_measure_unit_merger.rb

```diff
@@ -14,29 +18,35 @@ def call
   private
 
   def delta_measure_units
-    clean_unhandled_measure_units do
-      uk_filtered_commodity.applicable_measure_units.merge(xi_filtered_commodity.applicable_measure_units)
-    end
+    applicable_units =  uk_filtered_commodity.applicable_measure_units.merge(xi_filtered_commodity.applicable_measure_units)
```

Comment:
thiis feels a bit hard to follow, two long names with `.merge` tucked away in the middle - how about

```ruby
uk_units = uk_filtered_commodity.applicable_measure_units
xi_units  = xi_filtered_commodity.applicable_measure_units
applicable_units = uk_units.merge(xi_units)

clean_and_dedupe_measure_units applicable_units
```

Reply:
Yeah, happy to help with readability! I came back to this and had to remind myself what was going on so appreciate the suggestion